staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -182,6 +186,21 @@ type SharedInformer interface {
 	SetWatchErrorHandler(handler WatchErrorHandler) error
 }
 
+// DoneChannel is a type alias for a channel that closes when an informer is stopped.
+type DoneChannel <-chan struct{}
+
+// StopOnListErrorFunc is a type alias for function that takes
+// an error from the reflector's List call and
+// and determines whether or not to stop the reflector (and corresponding controller and shared informer).
+type StopOnListErrorFunc func(error) bool
```

Comment:
inside reflector, this function is called on the err returned from ListAndWatch, the err does not necessarily List error, it could be Watch error too. In this case, the StopOnListErrorFunc is a misnomer.

Reply:
Yea I that is correct, but `StopOnListAndWatchErrorFunc` seemed like a mouthful. What do you think about `StopOnReflectorErrorFunc`? It's short, but isn't exactly descriptive if you're wondering why the reflector would error out. If you think `StopOnListAndWatchErrorFunc` is better I'm happy to go that route too, open to whatever you think the best name would be.

Reply:
StopOnListErrorFunc is a bit misleading, StopOnListAndWatchErrorFunc is too long, what about StopOnErrorFunc? and add comment to explain that the error func is called inside reflector, and it is checked for both List and Watch errors.

Reply:
Sounds good, change made and comment added.

Reply:
changed throughout the PR